pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -31,59 +31,108 @@ import (
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pilot/pkg/networking/plugin"
 	"istio.io/istio/pilot/pkg/networking/util"
-	"istio.io/istio/pilot/pkg/proxy/envoy/v1"
 	"istio.io/istio/pkg/log"
 )
 
-// Plugin is a mixer plugin.
-type Plugin struct{}
+type mixerplugin struct{}
+
+type attribute = *mpb.Attributes_AttributeValue
+
+type attributes map[string]attribute
+
+const (
+	// UnknownDestination is set for cases when the destination service cannot be determined.
+	// This is necessary since Mixer scopes config by the destination service.
+	UnknownDestination = "unknown"
+
+	//mixerPortName       = "grpc-mixer"
+	// defined in install/kubernetes/helm/istio/charts/mixer/templates/service.yaml
+	mixerPortNumber = 9091
+
+	//mixerMTLSPortName   = "grpc-mixer-mtls"
+	mixerMTLSPortNumber = 15004
+
+	// mixer filter name
+	mixer = "mixer"
+)
 
 // NewPlugin returns an ptr to an initialized mixer.Plugin.
 func NewPlugin() plugin.Plugin {
-	return Plugin{}
+	return mixerplugin{}
```

Comment:
update function comment

Reply:
Why do we need a factory and a public constructor?